test/Streamly/Test/Data/Parser.hs

```diff
@@ -275,6 +275,22 @@ takeProperties =
 --                         where
 --                             list_length = Prelude.length ls
 
+sliceBeginWith :: Property
+sliceBeginWith =
+    forAll (listOf (chooseInt (min_value, max_value))) $ \ls ->
+        case S.parse (P.sliceBeginWith predicate FL.toList) (S.fromList ls) of
+            Right parsed_list -> if not $ Prelude.null ls
+                                 then
+                                    checkListEqual parsed_list $
+                                     if predicate (head ls)
+                                     then head ls : Prelude.takeWhile (not .
+                                     predicate) (tail ls)
```

Comment:
You can take stuff common using `let` and `where`. A continuous statement should be indented with 4 spaces if it is on the next line. In general, you don't want to split an expression.
`(not . predicate)` should be on the same line.
Something like this maybe,
```suggestion
                                     then head ls 
                                              : Prelude.takeWhile 
                                                    (not . predicate)
                                                    (tail ls)
```

Reply:
```
head ls : Prelude.takeWhile 
    (not .predicate) (tail ls)
```
Is this okay?

Reply:
An expression should ideally be in 1 line. If it spans 2 lines, it should be indented accordingly.

`Prelude.takeWhile  (not .predicate) (tail ls)` is an expression.

So, something like
```
head ls
    : Prelude.takeWhile (not .predicate) (tail ls)
```
`:` is on the next line as it is an infix operator. And we have some coding guidelines. They need not be strictly followed though.
```
head ls :
    Prelude.takeWhile (not .predicate) (tail ls)
```
This is also fine.
```
head ls :
    Prelude.takeWhile (not .predicate) (tail ls)
```

Reply:
Okay

Reply:
I have used let to intoduce a binding